### PR TITLE
[#7597] fix: fix EntityType short name duplicate issue

### PR DESCRIPTION
**Pull request by liunaijie**
### What changes were proposed in this pull request?

The `EntityType ` class has short name variable, but it's duplicated.  Table and Tag both used `ta`. 
And the short name is not used anymore.
This pr is try to clean up related code.

### Why are the changes needed?

Fix: #7597
### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Use existing tests.

### CODE REVIEW

File: core/src/main/java/org/apache/gravitino/Entity.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ enum EntityType {
     USER("us", 7),
     GROUP("gr", 8),
     ROLE("ro", 9),
-    TAG("ta", 10),
+    TAG("tag", 10),
```
Comment (justinmclean) [inline core/src/main/java/org/apache/gravitino/Entity.java]: Given all the other are two letters I'd go with "tg"

**Comment by justinmclean** [comment]
@liunaijie can you update the above PR text

**Comment by jerryshao** [comment]
IIRC, short name is not used anymore and can be removed, @yuqi1129 can you confirm and review this?

**Comment by yuqi1129** [comment]
> IIRC, short name is not used anymore and can be removed, @yuqi1129 can you confirm and review this?

Yeah, since `KvEntityStore` was no longer supported since 0.5,  and have no plan to support kv format storage in the near future, I believe we can remove the short name from the code.

**Comment by liunaijie** [comment]
> > IIRC, short name is not used anymore and can be removed, @yuqi1129 can you confirm and review this?
> 
> Yeah, since `KvEntityStore` was no longer supported since 0.5, and have no plan to support kv format storage in the near future, I believe we can remove the short name from the code.

Thans for your review, Can I remove the short name related code in this pr? 
I am not family with this project, just saw this good first issue https://github.com/apache/gravitino/issues/7597 and want to solve it.

**Comment by justinmclean** [comment]
It appears that type.getShortName() is still used in the code. It may be that this code is no longer needed, @yuqi1129?

**Comment by liunaijie** [comment]
> It appears that type.getShortName() is still used in the code. It may be that this code is no longer needed, @yuqi1129?

My bad, I am not totally remove the related code, The error is in `KvEntityStore ` class's `toString()` method. 
> Yeah, since KvEntityStore was no longer supported since 0.5

**Comment by justinmclean** [comment]
You'll also need to update some of the TestEntityCacheKey tests that depend on the removed code.

File: core/src/main/java/org/apache/gravitino/cache/EntityCacheKey.java [hunk 2]
```diff
@@ -138,7 +138,7 @@ public int hashCode() {
    */
   @Override
   public String toString() {
-    String stringExpr = identifier.toString() + ":" + type.getShortName();
+    String stringExpr = identifier.toString() + ":" + type.toString();
```
Comment (liunaijie) [inline core/src/main/java/org/apache/gravitino/cache/EntityCacheKey.java:141]: In the `CaffeineEntityCache` class.  when addCache, it will put data to `cacheData` (this struct store the data) and add key to `cacheIndex` struct (use EntityCache,toString as key).

So if without type suffix, there will be a edge case. like this UT (https://github.com/apache/gravitino/blob/main/core/src/test/java/org/apache/gravitino/cache/TestCaffeineEntityCache.java#L229) . cache store 2 entities, but the size return 1.

Update here use `type.toString` as suffix

 Refer: https://github.com/apache/gravitino/blob/main/core/src/main/java/org/apache/gravitino/cache/CaffeineEntityCache.java#L295

**Comment by yuqi1129** [comment]
@liunaijie 
Please take time to modify the PR descriptions, thanks.

File: core/src/main/java/org/apache/gravitino/Entity.java [hunk 3]
```diff
@@ -58,40 +58,27 @@ public interface Entity extends Serializable {
...
+    COLUMN(4),
+    FILESET(5),
+    TOPIC(6),
+    USER(7),
+    GROUP(8),
+    ROLE(9),
+    TAG(10),
+    MODEL(11),
+    MODEL_VERSION(12),
+    AUDIT(65534);
```
Comment (jerryshao) [inline core/src/main/java/org/apache/gravitino/Entity.java]: I think the index number in enum is also not used anymore, and can be removed.

**Review by justinmclean — APPROVED** [review]
Thanks for fixing this, especially as it was more work than originally expected.
